### ISSUE TITLE
Producer.committableSink: offsets not committed, when MultiMessage is empty

### ISSUE THREAD

**Issue by herzrasen** [akka/alpakka-kafka#1200]
Hi,

is it the desired behaviour that when using `Producer.committableSink` in combination with an empty `MultiMessage` that the offset is not committed to Kafka? 

More specifically in `CommittingProducerSinkStage`:
```
      case multiMsg: MultiMessage[K, V, Committable] =>
        val size = multiMsg.records.size
        awaitingProduceResult += size
        awaitingCommitResult += 1
        val cb = new SendMultiCallback(size, multiMsg.passThrough)
        for {
          record <- multiMsg.records
        } producer.send(record, cb)
```

The `SendMultiCallback` is not sent, when `multiMsg.records` is empty. The offset is committed deep down in `collectOffset`

```
  private final class SendMultiCallback(count: Int, offset: Committable) extends Callback {
    private val counter = new AtomicInteger(count)

    override def onCompletion(metadata: RecordMetadata, exception: Exception): Unit =
      if (exception == null) {
        if (counter.decrementAndGet() == 0) collectOffsetMultiCb.invoke(count -> offset)
      } else sendFailureCb.invoke(count -> exception)
  }

  // ---- Committing
  private val collectOffsetCb: AsyncCallback[Committable] = getAsyncCallback[Committable] { offset =>
    awaitingProduceResult -= 1
    collectOffset(offset)
  }

  private def collectOffset(offset: Committable): Unit =
    if (updateBatch(offset)) commit(BatchSize)
    else if (isClosed(stage.in) && awaitingProduceResult == 0L) commit(UpstreamClosed)

  private def commit(triggeredBy: TriggerdBy): Unit = {
    if (offsetBatch.batchSize != 0) {
      log.debug("commit triggered by {} (awaitingProduceResult={} awaitingCommitResult={})",
                triggeredBy,
                awaitingProduceResult,
                awaitingCommitResult)
      val batchSize = offsetBatch.batchSize
      offsetBatch
        .commitInternal()
        .onComplete(t => commitResultCB.invoke(batchSize -> t))(materializer.executionContext)
      offsetBatch = CommittableOffsetBatch.empty
    }
    scheduleCommit()
  }
```

**Comment by ennru** [comment]
Thank you for bringing this up.
That is an oversight. An empty multi-message should behave like a pass-through message and the offset should be committed.
Would you be in a position to suggest the necessary changes in a Pull Request?

**Comment by herzrasen** [comment]
Yes, I will have a look at it and create a PR in the next few days.

**Comment by ennru** [comment]
Fixed with #1201
